stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -793,6 +793,30 @@ function dot(x::AbstractArray, y::AbstractArray)
     s
 end
 
+"""
+    dot(x, A, y)
+
+Compute the generalized dot product `x' * A * y` between two vectors `x` and `y`.
+For complex vectors, the first vector is conjugated.
+"""
+function dot(x::AbstractVector, A::AbstractMatrix, y::AbstractVector)
+    m = checksquare(A)
```

Comment:
Is there a reason to restrict to square matrices? We only need `x'*A*y` to make sense.

Reply:
No reason, I was a bit mislead by the structured matrix cases, and obviously unable to count dimensions correctly. 😄